src/flint/test/test_all.py

```diff
@@ -2834,6 +2834,44 @@ def setbad(obj, i, val):
         if type(p) == flint.fq_default_poly:
             assert raises(lambda: p.integral(), NotImplementedError)
 
+        # resultant checks.
+
+        if is_field and characteristic == 0:
+            # Check that the resultant of two cyclotomic polynomials is right.
+            # See Dresden's 2012 "Resultants of Cyclotomic Polynomials"
+            for m in range(1, 50):
+                for n in range(m + 1, 50):
+                    a = flint.fmpz_poly.cyclotomic(m)
+                    b = flint.fmpz_poly.cyclotomic(n)
```

Comment:
These tests are running in a loop over all polynomial types but don't depend on the type.

I think it would be better just to move all of this out to a separate `test_poly_resultants` test function but still use `_all_polys` for the part that loops over the types.

Reply:
In general it would be better not to have such large test functions as this one.

Reply:
I mean that test_all_polys is too long.

Reply:
Ideally the test suite would have reasonably short test functions like `test_resultants` or something but where each test loops over all possible polynomial types. When a new polynomial type is added it should be possible to just add it to to the list in `_all_polys` and then update each test or add the methods that are needed for consistency with other types.